specifyweb/frontend/js_src/lib/components/Statistics/index.tsx

```diff
@@ -0,0 +1,599 @@
+          <div className="flex items-center gap-2">
+            <H2 className="text-2xl">{statsText('collectionStatistics')}</H2>
+            <span className="-ml-2 flex-1" />
+            {isEditing && (
+              <Button.Red onClick={(): void => setLayout(defaultLayout)}>
+                {commonText('reset')}
+              </Button.Red>
+            )}
+            <Button.Green
+              onClick={(): void =>
+                setState(
+                  isEditing
+                    ? { type: 'DefaultState' }
+                    : {
+                        type: 'EditingState',
+                        addingItem: undefined,
+                      }
+                )
+              }
+            >
+              {isEditing ? commonText('save') : commonText('edit')}
+            </Button.Green>
+          </div>
+
+          <div className="grid grid-cols-[repeat(auto-fill,minmax(20rem,1fr))] gap-4">
+            <CategoriesBoxes
+              goToEditingState={(categoryindex): void =>
+                setState({
+                  type: 'EditingState',
+                  addingItem: {
+                    pageIndex: activePageIndex,
+                    categoryIndex: categoryindex,
+                  },
+                })
+              }
+              isEditing={isEditing}
+              pageLayout={layout[activePageIndex]}
+              statsSpec={statsSpec}
+              onClick={undefined}
+              onRemove={
+                isEditing
+                  ? (categoryIndex, itemIndex): void =>
+                      setLayout(
+                        replaceItem(layout, activePageIndex, {
+                          ...layout[activePageIndex],
+                          categories: replaceItem(
+                            layout[activePageIndex].categories,
+                            categoryIndex,
+                            {
+                              ...layout[activePageIndex].categories[
+                                categoryIndex
+                              ],
+                              items: removeItem(
+                                layout[activePageIndex].categories[
+                                  categoryIndex
+                                ].items,
+                                itemIndex
+                              ),
+                            }
+                          ),
+                        })
+                      )
+                  : undefined
+              }
+            />
+          </div>
+        </div>
+      </div>
+    </>
+  );
+}
+
+function AddStatDialog({
+  onClose: handleClose,
+  onAdd: handleAdd,
+  defaultLayout,
+  statsSpec,
+}: {
+  readonly onClose: () => void;
+  readonly onAdd: (item: CustomStat | DefaultStat) => void;
+  readonly defaultLayout: StatLayout;
+  readonly statsSpec: IR<
+    IR<{
+      readonly label: string;
+      readonly items: StatCategoryReturn;
+    }>
+  >;
```

Comment:
Could you create a typescript type for this and use it everywhere, rather than specifying this long type repeatedly several times?